homeassistant/components/obihai/sensor.py

```diff
@@ -136,8 +137,9 @@ def update(self):
 
         services = self._pyobihai.get_line_state()
 
-        if self._service_name in services:
-            self._state = services.get(self._service_name)
+        if services is not None:
```

Comment:
Do you need this check if you are no longer creating the sensor?

Reply:
Yea I was getting a failure because certain devices that do not support the feature were failing to update.  So just skipping it if there is nothing there to update.